### ISSUE TITLE
[Due for payment 2025-02-19] [$500] [MacOS Chrome / iOS Chrome] Chrome Browser Has Mistaken Form Fields For Inputs in Expensify Chat

### ISSUE THREAD

**Issue by yuwenmemon** [Expensify/App#47939]
**Version Number:** 9.0.23-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Caught during [CVP Testing](https://github.com/Expensify/Expensify/issues/421286)
**Email or phone of affected tester (no customers):** yuwenis+alice1@gmail.com
**Logs:** N/A
**Expensify/Expensify Issue URL:** N/A
**Issue reported by:** @yuwenmemon 
**Slack conversation:** https://expensify.slack.com/archives/C036QM0SLJK/p1724431920271019

## Issue
1. On the same account, go to the chat with Concierge on mobile web
2. Click on the text input, your phone keyboard should pop up

### Expected Result:
- The keyboard should show on mobile chrome without any form input helpers (i.e. the key, creditcard, and location options above the keyboard)

### Actual Result:
-  Chrome thinks the input is a form field and thus presents some form field shortcuts above the keyboard:
<img src="https://github.com/user-attachments/assets/f6e88fb4-f699-43df-a6d8-06eab3898067" alt="IMG_5393" width="300"/>

## Workaround:
N/A

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021846212488219658542</li>
        <li>Upwork Job ID: 1846212488219658542</li>
        <li>Last Price Increase: 2024-11-19</li>
<li>Automatic offers: </li>
<ul>
<li>allgandalf | Reviewer | 104995463</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @RachCHopkins</details>

**Comment by klajdipaja** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chrome browser on mobile tries to autocomplete some input fields like the name in the signup process or in the profile, the merchant when adding expenses etc and in doing so it shows options like Credit Card autofill.
NOTE. I was unable to reproduce this in the Chat with concierge.

### What is the root cause of that problem?
Autocomplete is not disabled in `components/TextInput` even though we are treating it as disabled by not propagating the property in the `components/Form/InputWrapper` component. Since it's not disabled Chrome uses the label to try and understand how to do that autofill.

### What changes do you think we should make in order to solve the problem?
Before passing the props by the wrapper component in:
https://github.com/Expensify/App/blob/0c8455280c738a5db596f34409a0a3177e682e7f/src/components/TextInput/index.tsx#L58-L59

We should disable it explicitly by adding:

``` typescript
            autoComplete="off"
            // eslint-disable-next-line react/jsx-props-no-spreading 
            {...props}
 ```

That would still allow for the `props` to add an autocomplete option in the wrapper component for special use-cases of the input field.

### What alternative solutions did you explore? (Optional)

**Comment by anmurali** [comment]
Cannot reproduce the first issue 
<img width="1434" alt="image" src="https://github.com/user-attachments/assets/864506f2-acd4-4dd1-a798-0c0af9956c2d">

I can reproduce the second

**Comment by trjExpensify** [comment]
@allgandalf what do you think of the proposals?

**Comment by trjExpensify** [comment]
Thanks!

**Comment by klajdipaja** [comment]
@allgandalf I was unable to reproduce the first issue, only the second one.
As for my proposal you are right I tried again and it did not provide the expected fix and with all my research all I could find is that this is a default behavior of Chromium browsers. 
The root cause to me seems to be a "bug" or maybe default behavior in Chrome that for every input no matter if you have requested autocompletion or not, the autocompletion bar with the three icons show up. When autocomplete is specified to some type than in addition to these three default icons you could also see a list of suggestions.

There's a long list of hacks on this gist https://gist.github.com/niksumeiko/360164708c3b326bd1c8 that some of the users report as useful but non actually work to remove the whole autofill bar and to be honest I don't think we should aim to go in that direction.
I see three options here:
1. Set autocomplete values appropriately when possible so that the autofill bar becomes meaningful for most inputs. At the moment we are not doing this for 90% of inputs, even email fields do not have the autocomplete set to email which would help users fill the data.
2. Ignore this issue as a chrome behaviour and avoid building workarounds that might become unnecessary in a feature chrome update.
3. Build a workaround. The only workaround I have found for this is to make the input a text-area which is why in some fields like the chat we do not get the autofill bar. That could be done by adding 
```
    maxLength={25}
    multiline={true}
    numberOfLines={1}
```.
I can prepare a proposal for this if you think that would be the direction you want to go.

**Comment by Skalakid** [comment]
Hello, I'm Michał from SWM, and I will help investigate this issue. I can't reproduce this issue, is it still present in the app? After creating a new account I followed all the steps specified in the issue description, and I can't see the autofill bar above the keyboard in the chat with the Concierge

https://github.com/user-attachments/assets/4b2e18d5-df21-4127-838f-8acef7b4fbf1

**Comment by stephanieelliott** [comment]
Just noting here that I am also unable to repro the first issue

**Comment by Skalakid** [comment]
Hello, I was OOO recently, but @BartoszGrajdek has found the solution for this issue. [Here](https://github.com/Expensify/react-native-live-markdown/pull/506), I created the PR that contains the fix

**Comment by Skalakid** [comment]
@allgandalf Yes, I was able to reproduce the second issue on mobile Safari. Working on the fix


<img src="https://github.com/user-attachments/assets/53ce8e24-07cd-4f91-9aa0-e24f89e0a0bd" width="200px" />

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Skalakid** [comment]
@allgandalf, no updates from my side. I started the investigation. However, I haven't been able to find a solution yet. Now I have to prioritize some other tasks, but once I finish them, I will come back to this issue

**Comment by Skalakid** [comment]
@allgandalf I will ask on our internal chanel

**Comment by Skalakid** [comment]
@allgandalf Sorry, but currently no one form our team have any capacity to investigate this issue. Let's open it for contributor proposals

**Comment by skyfloyd** [comment]
I think the first issue can be fixed by correcting `autocomplete` property on the inputs.
Second issue can be solved by using `KeyboardAwareScrollView`
My Upwork profile is https://www.upwork.com/freelancers/mhera

**Comment by skyfloyd** [comment]
Contributor details
Your Expensify account email: mher.aghabalyan@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/mhera

Meanwhile, I will try to run this application on local. It will take some time

**Comment by skyfloyd** [comment]
Hi guys, I have notice that to run app on android I have to be registered in mapbox.com which requires my credit card details. Do you have any test mapbox.com account as I will not like to provide my credit card details to a third-party application?

**Comment by Frank-Masciopinto** [comment]
Adjust the name attribute: Sometimes, Chrome uses the name attribute to infer the type of data being entered. 

Ensure the autocorrect attribute is set to off for fields where you don’t want any additional suggestions:

**Comment by trjExpensify** [comment]
@anmurali as an onboarding related bug, should we put this in the #convert project board now #wave-collect is done?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by skyfloyd** [comment]
I have sent Upwork proposal, but it seems no one cares ))

**Comment by yuwenmemon** [comment]
Still waiting on proposals as per @allgandalf

**Comment by anmurali** [comment]
Increasing the bounty didn't do it. @yuwenmemon we might need to fix this internally.

**Comment by anmurali** [comment]
Rotating the label so we can assign another BZ member while I am OOO

**Comment by yuwenmemon** [comment]
Asking in Slack: https://expensify.slack.com/archives/C04878MDF34/p1731966318908489

To reduce confusion I'm removing the first issue from the OP since no one is able to reproduce anymore. But for posterity it was the following:

> **There are two issues that are similar.**
> 
> ## First Issue
> ### Action Performed:
> 1. Create an account on NewDot
> 2. Choose `Get paid back from my employer` from the intent selector
> 3. On the form to enter one's first and last name, focus the first input
> 
> ### Expected Result:
> - Chrome autofill should suggest your name, without implying that it's a credit card form
> 
> ### Actual Result:
> - Chrome autofill shows your name in the context of a Credit Card input form. 
> ![screenshot_2024-08-23_at_12 35 56___pm_720](https://github.com/user-attachments/assets/e9daaf23-fe12-4ed8-993f-f7e94f74f007)

**Comment by RachCHopkins** [comment]
Thanks for that @yuwenmemon, I just spent far too long trying to work out what issue 2 was when I could only see 1!

**Comment by yuwenmemon** [comment]
@SzymczakJ Expressed interest in picking this one up!

**Comment by SzymczakJ** [comment]
Hey! I’m Jakub Szymczak from Software Mansion, an expert agency, and I’d like to work on this issue!

**Comment by SzymczakJ** [comment]
When I tested this issue on real device(as I thought it would give me more insight) I noticed it's not reproducible on real device. 
So here comes my question: **should we be fixing a bug that only occurs on mWeb iOS simulator?** This might be even caused by some solely simulator internals bug. WDYT @yuwenmemon @allgandalf?

**Comment by yuwenmemon** [comment]
Interesting. I first saw the issue on a real iOS device.

I'm unsure the environment where others were able to reproduce but looking at the screenshots it's both on Safari and Chrome?

> should we be fixing a bug that only occurs on mWeb iOS simulator?

I would say it depends on what we need to do to fix it. I always thought it was some weird component/html thing getting misinterpreted by the browser.

**Comment by SzymczakJ** [comment]
I can only reproduce it on simulator, but I'm going to use this to investigate.

**Comment by SzymczakJ** [comment]
For investigation purposes: What's the iOS version and Safari version on which you managed to reproduce it? I was trying to reproduce on like 5 different iPhones and didn't manage to do it.
Also do you have some specifc setup, like username saved in apple keychain, or some other iPhone/account config?

**Comment by SzymczakJ** [comment]
Here's the investigation outcome:
tl:dr This is a Safari bug, which I managed to reproduce only on simulator. I've filed in a bug report to Safari and because I don't see any way of fixing this internal safari problem, then we should put it on hold and wait for safari to fix it.

To make things clear here's the bug that I'm talking about(after writing to 'login username text input' the password autofill suggestion is also appearing in all other text inputs):

https://github.com/user-attachments/assets/65d840e4-0bd3-424b-8106-4457b6fe2b53

**I'm 95% sure this is a Safari/simulator bug, because of the following reasons**:
1. I created minimal reproduction app(https://github.com/SzymczakJ/text-input-type-repro) in which I can reproduce the bug behaviour. So this bug is not connected to Expensify directly and is not caused by some internal Expensify config.
2. It's not a bug that can be solved by setting proper props to `TextInput`, I've tried setting/resseting all of them.
3. I managed to reproduce the same behaviour on Apple website 😃, so I guess this is just how safari works, with mobile keyboards:

https://github.com/user-attachments/assets/30dc241f-9f66-4a10-b8ac-874ce934c7df

**I've reported a bug on Safari bug report Website, let's see if they respond. If you want me to investigate further, then please provide me more information about the iOS and Safari versions on which this bug is repoducible on real devices, because I'm running out of ideas and for me this bug only happens for simulators(I've tested 5 different iPhones).**
This bug will only happen in case when user logs into app and then uses the app right away. Cases like going back to app, when already logged in, or resetting the safari fix the bug, so the buggy behaviour will happen in minority of cases. Because of that **I'm in favour of putting this on hold and wait for Safari to solve the problem as we don't have any way of changing Safari internal behaviour**. WDYT @allgandalf @yuwenmemon

**Comment by RachCHopkins** [comment]
@allgandalf @yuwenmemon, what do you think of that plan? Makes sense to me.

**Comment by yuwenmemon** [comment]
> I was able to reproduce on real device on Safari

@allgandalfare you able to share the device details you were able to reproduce this on?

In any case, if this seems like a simulator issue only and we've had trouble reproducing then maybe we can just close this out TBH.

**Comment by davidcardoza** [comment]
I was able to reproduce this. When going through the CVP, once I get to the step to add the first/last name on Chrome mobile web, the Last Name input field starts suggesting credit cards stored in my Chrome account.
![IMG_CDE73D085CDF-1](https://github.com/user-attachments/assets/ba01085a-e865-44d4-9122-9c07fc50cc88)

**Comment by RachCHopkins** [comment]
Ok, well it's a real issue then! And not just a safari bug. @allgandalf where to next?

**Comment by SzymczakJ** [comment]
I have two issues on me https://github.com/Expensify/App/pull/53389 and https://github.com/Expensify/App/issues/52317, one close to being merged and the second one in WIP state. Is it okay with you, if I finish these tasks first, before moving on to this issue(keep in mind that SWM devs have 2 weeks of OOO during the Christmass time, so everything can take me more time, probably 3-4 weeks).
I could also ask other SWM developers, if they have spare time to take this task. WDYT @RachCHopkins @allgandalf @yuwenmemon

**Comment by SzymczakJ** [comment]
Unfortunately everybody on SWM is busy and cannot take this issue at the moment.
Other option would be to again open it up to externals and if nobody takes it I will work on it, when I finish my Search issues, that I listed above.
I'm sorry that I cannot take this bug right away, but I just feel that Search feature has more priority than this bug. WDYT

**Comment by quinthar** [comment]
Still happening, note that it's prompting me to enter a credit card, on Android Chrome.

![image](https://github.com/user-attachments/assets/798673cc-80f4-48c4-9afe-09945d68bb6a)

**Comment by SzymczakJ** [comment]
I'm back from OOO, I'm planning to start working on it this week I just need to put up a PR for https://github.com/Expensify/App/issues/52317. I'll try to make is as fast as possible to take up this task again.

**Comment by yuwenmemon** [comment]
Thanks @SzymczakJ!

**Comment by SzymczakJ** [comment]
I'll try to take a look at it today, the PR that I was talking about is almost ready.

**Comment by SzymczakJ** [comment]
FYI I started to investigate this, I will split my work time between this issue and finishing https://github.com/Expensify/App/pull/55078(which soon will be in review stage, so I will have more time for this bug). 
I'll publish my investigation progress every one/two days.

**Comment by SzymczakJ** [comment]
After two days of investigating this, this is what I’ve found:
1. Chrome Android bug(the autocomplete bar showing up on every input keyboard)

<img src="https://github.com/user-attachments/assets/92314381-a6e6-4afa-8c29-8f6f46bdaf35" width="250" height="500">

 is well known and we cannot really fix it as it’s on Chrome’s side(reference links: [link 1](https://support.google.com/chrome/thread/167858494/password-bar-above-keyboard-on-android-phone-always-there?hl=en), [link 2](https://stackoverflow.com/a/73466347)), but there’s one workaround that I’ve published in the PR above. This workaround won’t work if we need some specific type of inputMode, but it should solve most of the cases like onboarding inputs that show us the card autocomplete, so also card suggestions should not show.

2. The Chrome iOS bug(which is shown in the issue description)

<img src="https://github.com/user-attachments/assets/4e947592-150f-4c90-8df5-066933b5b14c" width="250" height="500">

 is also well known issue and also out of our reach as it’s caused by Apple itself(ref: [link1](https://support.google.com/chrome/thread/21001996/chrome-shortcuts-above-keyboard-on-ios?hl=en), [link2](https://stackoverflow.com/questions/61788900/how-to-disable-ios-chrome-password-credit-card-autofill)). After trying out every prop of text input and long search over the internet I don’t see any workaround for this, but the same bug also happens on all popular sites like amazon of facebook, so I think nobody can solve this and we’ll just have to live with it.

I think we could just push the Chrome Android fix as it’s the most we can do in this case. WDYT @yuwenmemon @allgandalf

**Comment by SzymczakJ** [comment]
I'll upload videos and test more, when @yuwenmemon gives us green light 🟢

**Comment by yuwenmemon** [comment]
@allgandalf @SzymczakJ Better than nothing! Let's move forward but just be sure to effectively document that this is a workaround in the code where we implement this.

**Comment by RachCHopkins** [comment]
@allgandalf can you do the checklist please?

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @SzymczakJ no payment due to contractor status
- Contributor+: @allgandalf paid $250 via Upwork

Upwork job [here](https://www.upwork.com/jobs/~021846212488219658542)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed.
